File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    List<TbBrand> findAll();
    //分页查询数据
    PageResult findPage(int pageNum,int pageSize);

    void  add(TbBrand brand);

    //获取当前的Brand
    TbBrand findOne(Long id);

    void update(TbBrand brand);

    void dele(Long[] ids);

    PageResult findSearch(TbBrand brand,int pageNum,int pageSize);

    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }

    this.findPage=function (page,rows) {
        return  $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    }

    this.add=function (entity) {
        return $http.post('../brand/add.do',entity);
    }

    this.update=function (entity) {
        return $http.post('../brand/update.do',entity);
    }

    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    }

    this.dele=function (ids) {
        return $http.get('../brand/dele.do?ids='+ids);
    }

    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }
});